desktop/apps/consignments2/client/actions.js

```diff
@@ -328,3 +404,72 @@ export function updateSubmission (submission) {
     }
   }
 }
+
+export function uploadImageToConvection (filePath, fileName) {
+  return async (dispatch, getState) => {
+    try {
+      const token = await fetchToken()
+      const { submissionFlow: { submission } } = getState()
+      const geminiToken = last(filePath.split('/')).split('%2F')[0] // FIXME get the gemini token another way
+      const inputs = {
+        submission_id: submission.id,
+        gemini_token: geminiToken
+      }
+      await request
+        .post(`${sd.CONVECTION_APP_URL}/api/assets`)
+        .set('Authorization', `Bearer ${token}`)
+        .send(inputs)
+
+      dispatch(uploadImageToGemini(submission.id, filePath, fileName))
+    } catch (err) {
+      dispatch(updateError('Unable to upload image.'))
+      dispatch(stopProcessingImage(fileName))
+      console.error('error!', err)
+    }
+  }
+}
+
+export function uploadImageToGemini (submissionId, sourceUrl, fileName) {
+  return async (dispatch, getState) => {
+    try {
+      const inputs = {
+        entry: {
+          template_key: sd.CONVECTION_GEMINI_APP,
+          source_url: sourceUrl,
+          metadata: {
+            id: submissionId,
+            _type: 'Consignment'
+          }
+        }
+      }
+      await request
+        .post(`${sd.GEMINI_APP}/entries.json`)
+        .set('Authorization', `Basic ${encode(sd.CONVECTION_GEMINI_APP, '')}`)
+        .send(inputs)
+      dispatch(stopProcessingImage(fileName))
+    } catch (err) {
+      dispatch(updateError('Unable to process image.'))
+      dispatch(stopProcessingImage(fileName))
+      console.error('error!', err)
+    }
+  }
+}
+
+// helpers
+async function fetchToken () {
+  const {
+    body: {
+      token
+    }
+  } = await request
+              .post(`${sd.API_URL}/api/v1/me/token`)
+              .set('X-ACCESS-TOKEN', sd.CURRENT_USER.accessToken)
+              .send({ client_application_id: sd.CONVECTION_APP_ID })
+  return token
+}
+
+function encode (key, secret) {
+  if (window.btoa) {
+    return btoa(unescape(encodeURIComponent([key, secret].join(':'))))
+  }
```

Comment:
Ah, good catch. I'll remove the `window.btoa` check since it's not really doing much (i think I added it while I was figuring out a test), but if this fails it'll get caught in the `catch` block above and show something like: 
![image](https://cloud.githubusercontent.com/assets/2081340/26262306/571ffd5a-3ca2-11e7-8430-bd24ab236291.png)

I'll put a note to look into whether this function is cross-browser compatible (looks like it's fine in all modern browsers) to see if we should be using a library.